app/loader.cpp

```diff
@@ -263,3 +303,22 @@ QList<QgsExpressionContextScope *> Loader::globalProjectLayerScopes( QgsMapLayer
   scopes << QgsExpressionContextUtils::layerScope( layer );
   return scopes;
 }
+
+void Loader::setActiveLayerFromName( QString layerName ) const
+{
+  mLayersProxyModel.invalidate();
+  mActiveLayer.setActiveLayer(
+    mLayersProxyModel.layerFromName( layerName )
+  );
```

Comment:
I am wondering why we need layers proxy model here - why not just go through QgsProject's map layers and use the one with matching name?

Reply:
I understand the reason but .. would then more components be handling the same thing?

Reply:
well... it seems to me simpler to do just get layer from name using QgsProject::mapLayersByName() than going through some extra proxy model class... It would make the Loader design simpler if it knew nothing about some models and proxy models

Reply:
Cool, I did not know about `mapLayersByName` function :)

Reply:
updated